Packs/Slack/Integrations/SlackV3/SlackV3.py

```diff
@@ -2540,6 +2560,306 @@ def pin_message():
+    and events
+    """
+
+    channel_id = demisto.args()['channel_id']
+    limit = demisto.args().get('limit')
+    if limit == None:
+        limit = 100
+
+
+    conversation_id = demisto.args().get('conversation_id')
+    if conversation_id == None:
+        body = {
+            'channel': channel_id,
+            'limit': limit
+        }
+    else:
+        body = {
+            'channel': channel_id,
+            'oldest': conversation_id,
+            'inclusive': "true",
+            'limit': 1
+        }
+
+
+    raw_response = send_slack_request_sync(CLIENT, 'conversations.history', http_verb="GET", body=body)
+    messages = raw_response['messages']
+
+
+    if raw_response['ok'] == False:
+        error = raw_response['error']
+        return_error(f'You received the following error: {error}')
+
+
+    if type(messages) == dict:
+        messages = [messages]
+
+
+    if type(messages) == list:
+        context = []
+        for message in messages:
+            if 'subtype' not in message:
+                user_id = message['user']
+                body = {
+                    'user':user_id
+                }
+                user_details_response = send_slack_request_sync(CLIENT, 'users.info', http_verb="GET", body=body)
+                user_details = user_details_response['user']
+                if 'thread_ts' in message:
+                    entry = {
+                        'Type': message['type'],
+                        'Text': message['text'],
+                        'UserId': message['user'],
+                        'Name': user_details['name'],
+                        'FullName': user_details['real_name'],
+                        'TimeStamp': message['ts'],
+                        'HasReplies': 'Yes',
+                        'ThreadTimeStamp': message['thread_ts']
+                    }
+                    context.append(entry)
+                else:
+                    entry = {
+                        'Type': message['type'],
+                        'Text': message['text'],
+                        'UserId': message['user'],
+                        'Name': user_details['name'],
+                        'FullName': user_details['real_name'],
+                        'TimeStamp': message['ts'],
+                        'HasReplies': 'No',
+                        'ThreadTimeStamp': 'N/A'
+                    }
+                    context.append(entry)
+            elif 'subtype' in message and 'thread_ts' in message:
+                entry = {
+                    'Type': message['type'],
+                    'Text': message['text'],
+                    'UserId': message['username'],
+                    'Name': message['username'],
+                    'FullName': message['username'],
+                    'TimeStamp': message['ts'],
+                    'HasReplies': 'Yes',
+                    'ThreadTimeStamp': message['thread_ts']
+                }
```

Comment:
Could you please consider refactoring the code to eliminate the duplicate code?

Reply:
Done